### PR TITLE
Create SECURITY.md

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

Draft default security policy.  This is a requirement for ITN and needed in order for us to have a bug bounty program. 

By making the `SECURITY.md` file here in the `celestiaorg/.github` repo, it will enable the security policy org wide. 

Additionally the plan will be to enable security reporting on all the repos. From a permissions standpoint, the people that will be able to see these security advisories are:
- Repo Admins
- Org Owners
- Org Security team (this is a specific role in github)

I pulled much of the policy from other projects I'm familiar with, so very open to suggestions of additions and deletions. 

Walkthrough 

https://user-images.githubusercontent.com/15232757/212750597-77f8980c-341d-498a-93e7-f83957ab1d32.mp4

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: SECURITY.md
Comment (liamsi) [inline SECURITY.md:32]: Is this supposed to be the official comms channel between the reporting party and Celestia? That should always be security@celestia.org for the sake of simplicity IMO (unless agree with the party differently).
Celestia's official comms channels can be found on the homepage under community: https://celestia.org/

**Comment by musalbas** [comment]
>By making the SECURITY.md file here in the celestiaorg/.github repo, it will enable the security policy org wide.
Would this mean that all our repos would automatically have this security policy, or do we also have to copy this md file to every repo?

>Additionally the plan will be to enable security reporting on all the repos.
I think it might not be realistic to assume by default that all 100 repos in our org will have security reporting setup (is that a lot of overhead compared to assuming a security@ email?), but only the main repos, so the security policy could perhaps be defined with that in mind.

File: SECURITY.md
Comment (musalbas) [inline SECURITY.md:52]: Official Channels is undefined. Should that be "Official Communication Channels"?

File: SECURITY.md
Comment (musalbas) [inline SECURITY.md:69]: (as above)

File: SECURITY.md
Comment (MSevey) [reply SECURITY.md:5]: This is where a repo specific security.md file would make sense. So node for example can list their specific support versions which will be different from core, app, etc.

Comment (musalbas) [reply SECURITY.md:5]: Then perhaps we should not define what release versions are supported in this policy, but let each repo define it, and just say in case it's not defined, we recommend the most recent minor release?

File: SECURITY.md
Comment (MSevey) [reply SECURITY.md:22]: I can say that this applies to the Celestia Org github and all repos within that scope. 

As for infrastructure, I would include that by means of any infrastructure-as-code that we have. Such as ansible scripts for initializing servers and spinning up services. That will live in a github repo and so any security risks with our infrastructure would be reported there. 

That is how I would think about it.

**Comment by MSevey** [comment]
> By making the SECURITY.md file here in the celestiaorg/.github repo, it will enable the security policy org wide.

> Would this mean that all our repos would automatically have this security policy, or do we also have to copy this md file to every repo?

Made a little recording that hopefully clarifies cc @musalbas 


https://user-images.githubusercontent.com/15232757/212750536-9f30bb4f-84dd-40c3-a28d-91704888ba13.mp4

File: SECURITY.md
Comment (MSevey) [reply SECURITY.md:32]: This should be where we plan to communicate security updates publicly. 

So one piece will be that all communication about security reports would come from the `security@celestia.org` email, but then where do we want to say we will announce security updates? Twitter? Discord? Github?

Comment (MSevey) [reply SECURITY.md]: made a general update

Comment (musalbas) [reply SECURITY.md]: Notes:
1. just fyi: security@ is a Google group to receive emails, so it can't be the _sender_of emails
2. I think saying "celestia.org" is the official channel for announcements isn't good enough as that's too vague. I think there should probably be a concrete place for people to receive security advisories - would this be via the GitHub security advisories feature by default?

Comment (MSevey) [reply SECURITY.md]: > would this be via the GitHub security advisories feature by default

I don't think so, those are for reporting to us, not us reporting out.  Only owners, admins, and the security team have access to see those.

> just fyi: security@ is a Google group to receive emails, so it can't be the sender of emails afaik

You can actually set it as an alias, so you can send emails from a google group https://support.google.com/googlecloud/answer/10635789?hl=en

>  I think there should probably be a concrete place for people to receive security advisories

I agree, should we just pick one? Say discord?

File: SECURITY.md
Comment (MSevey) [reply SECURITY.md:5]: Yea makes, sense. I'll re-word to make that more clear.

File: SECURITY.md
Comment (musalbas) [reply SECURITY.md]: >I don't think so, those are for reporting to us, not us reporting out. Only owners, admins, and the security team have access to see those.

>I agree, should we just pick one? Say discord?

https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/publishing-a-repository-security-advisory
?

>You can actually set it as an alias, so you can send emails from a google group https://support.google.com/googlecloud/answer/10635789?hl=en

I'm unsure if it's good practice to be sending emails from security@ though; rather they should be attributed to an individual on the team. But I don't have a strong opinion.

File: SECURITY.md
Comment (liamsi) [inline SECURITY.md:23]: I wonder if we should include something about how we handle forks. Of which we have quite a few in our org 🤔

**Review by liamsi — COMMENTED** [review]
I think this is a great first step. Do not really have further comments. I think we should modify the security.md file of core and potentially other forks to clarify the scope that people can report on in these respectively.

File: SECURITY.md
Comment (MSevey) [reply SECURITY.md]: > https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/publishing-a-repository-security-advisory

TIL. then yea, we could use this.

Comment (MSevey) [reply SECURITY.md]: I can update once Jeremey reviews and include any additional updates

Comment (musalbas) [reply SECURITY.md]: Regarding the official communications channel - does GitHub also not have a built-in way to respond to security reports inside of GitHub, and if so should that be mentioned?

File: SECURITY.md
Comment (musalbas) [inline SECURITY.md:39]: that?

File: SECURITY.md
Comment (musalbas) [inline SECURITY.md:23]: needs updating - see google doc

File: SECURITY.md
Comment (musalbas) [reply SECURITY.md:39]: Sentence seems incomplete as it abruptly ends with "that"

File: SECURITY.md
Comment (adlerjohn) [inline SECURITY.md:5]: Nit, but it's weird that lines exceed 80 chars here but not lower down.

File: SECURITY.md
Comment (adlerjohn) [inline SECURITY.md:18]: link to section "Reporting a Vulnerability"

File: SECURITY.md
Comment (adlerjohn) [inline SECURITY.md:23]:
```suggestion
- 120 days have passed after you submitted the vulnerability
```

File: SECURITY.md
Comment (adlerjohn) [inline SECURITY.md:27]: ?

```suggestion
The scope of this security policy applies to the code repositories under the [Celestia Labs GitHub](https://github.com/celestiaorg) and any related infrastructure.
```

File: SECURITY.md
Comment (adlerjohn) [inline SECURITY.md:43]:
```suggestion
If you have suggestions on how this process could be improved please submit a pull request.
```

File: SECURITY.md
Comment (musalbas) [reply SECURITY.md:27]: @CelestiaOrg is not the Celestia Labs GitHub. It is the Celestia community GitHub.

Comment (adlerjohn) [reply SECURITY.md:27]: Ah then it should be
```suggestion
The scope of this security policy applies to the code repositories under the [Celestia community Github](https://github.com/celestiaorg) and any related infrastructure.
```

Comment (musalbas) [reply SECURITY.md:27]: IMO it should just be explicit and say the CelestiaOrg GitHub. There's no need to give CelestiaOrg an alias.

File: SECURITY.md
Comment (MSevey) [reply SECURITY.md:23]: this grammatically sounds weird with `whichever is sooner` imo.  I think it's fine as it.

**Review by adlerjohn — APPROVED** [review]
LGTM, but defer to other reviewers.

**Review by liamsi — APPROVED** [review]

**Pull request merged**
